sdk/java-sdk/src/main/scala/com/akkaserverless/javasdk/impl/action/ActionsImpl.scala

```diff
@@ -83,10 +84,14 @@ final class ActionsImpl(_system: ActorSystem, services: Map[String, ActionServic
           toProtocol(forward.metadata()))
         Future.successful(ActionResponse(ActionResponse.Response.Forward(response), toProtocol(sideEffects)))
       case AsyncEffect(futureEffect, sideEffects) =>
-        futureEffect.flatMap { effect =>
-          val withSurroundingSideEffects = effect.addSideEffects(sideEffects.asJava)
-          effectToResponse(withSurroundingSideEffects, anySupport)
-        }
+        futureEffect
+          .flatMap { effect =>
+            val withSurroundingSideEffects = effect.addSideEffects(sideEffects.asJava)
+            effectToResponse(withSurroundingSideEffects, anySupport)
+          }
+          .recover { case NonFatal(ex) =>
+            ActionResponse(ActionResponse.Response.Failure(Failure(description = ex.getMessage)))
```

Comment:
Will this become the error handling behavior as the entities? For them I can see that exceptions are turned into EntityException...

Reply:
In the entities it is always a stream of events/commands => responses, and that has a recover in the "far end" catching any failure happening inside of the stream: 
https://github.com/lightbend/akkaserverless-java-sdk/blob/main/sdk/java-sdk/src/main/scala/com/akkaserverless/javasdk/impl/eventsourcedentity/EventSourcedEntitiesImpl.scala#L132

So this is aligning the various action calls with how the entities behave.

Reply:
I think we should add error logging for all these just like for entities though.

Reply:
Ah, one thing that is different wrt is that for entities the exception is hidden and the returned message is "internal error" unless it is specifically an `EntityException`, so that's the solution there for not "leaking" impl details in the error. Maybe I should do that here as well?